torch/csrc/jit/export.cpp

```diff
@@ -649,11 +833,23 @@ void ScriptModuleSerializer::convertModule(
     torch::ParameterDef* param_def = module_def->add_parameters();
     convertParameter(elem.value(), param_def, /*is_buffer=*/false);
   }
-  for (const auto& elem : module.get_attributes()) {
-    if (elem.value().type->isSubtypeOf(TensorType::get())) {
-      torch::ParameterDef* param_def = module_def->add_parameters();
-      convertParameter(elem.value(), param_def, /*is_buffer=*/true);
+
+  if (module.get_attributes().size() > 0) {
+    Pickler p(*this);
+    p.start();
```

Comment:
would it be better to record names of attributes in the pickled file as a gigantic dictionary?

also, why each attribute is not a separate file? we do store each tensor separately, so it feels logical to do the same for attributes

Reply:
especially when we have objects, attributes can reference each other so they need to go into one gigantic file so that we can deserialize them with the same aliasing relationships.